tests/probes_test.go

```diff
@@ -286,3 +270,21 @@ func createProbeSpecification(period int32, initialSeconds int32, handler v12.Ha
 func isHTTPProbe(probe v12.Probe) bool {
 	return probe.Handler.HTTPGet != nil
 }
+
+func serverStarter(vmi *v12.VirtualMachineInstance, probe *v12.Probe, port int) {
+	if isHTTPProbe(*probe) {
+		tests.StartHTTPServer(vmi, port)
+	} else {
+		tests.StartTCPServer(vmi, port)
+	}
+}
+
+func attachProbeToPodByIp(pod *v1.Pod, IPFamily v1.IPFamily, probe *v12.Probe) (*v12.Probe, error) {
+	supportPodIP := libnet.GetPodIpByFamily(pod, IPFamily)
+	if supportPodIP == "" {
+		return nil, fmt.Errorf("pod's %s %s IP address does not exist", pod.Name, IPFamily)
+	}
+	livenessProbe := patchProbeWithIPAddr(probe, supportPodIP)
```

Comment:
why liveness probe ? this should be used also for readiness probes.

Reply:
Damm. Forgot to change that. Thanks for noticing.
 Done.